### DISCUSSION TITLE
smart contract benefits

### DISCUSSION THREAD

**Discussion by Moemen12** [Cyfrin/foundry-full-course-cu#2111] [Q&A]
hello, i have deployed my first smart contract but actually i didnt get what's the idea of using it ... i meant i didnt see **real use case** ...

**Comment by Saintdmixthephysicist** [comment]
actually i am also having such conflicts thinking about it as a whole, what's the entire purpose of it all

**Comment by EngrPips** [comment]
It means you can write any agreement you can think of and deploy it to the blockchain, where it is unchangeable and transparent to the whole world what the terms of the agreements are before they agree to it and use it

**Reply by Moemen12** [reply]
I know it's unchangeable there but what about some functionnality like store or addFavoritePerson , what's the purpose of using them locally? What should I do with that contract after deploying it  ..

**Reply by EngrPips** [reply]
We use a function inside of a contract to build the feature of the contract. How do you mean `what's the purpose of using them locally?.` After deploying the contract, you should continue with the course, and if you are serious about your career as a web3 developer, you will later build your own project. and In the long you might create a web3 product with all the knowledge you have gathered or maybe get employed to work for some web3 companies.
